### PR TITLE
chore: update tests to use 1.0 image and update compat note

### PR DESCRIPTION
**Pull request by g-awmalik**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: README.md [hunk 1]
```diff
@@ -10,10 +10,12 @@ This module consists of the following submodules:
 See more details in each module's README.
 
 ## Compatibility
-This module is meant for use with Terraform 0.13. If you haven't
-[upgraded](https://www.terraform.io/upgrade-guides/0-13.html) and need a Terraform
+This module is meant for use with Terraform 0.13+ and tested using Terraform 1.0+.
+If you find incompatibilities using Terraform `>=0.13`, please open an issue.
+
+If you haven't [upgraded][terraform-0.13-upgrade] and need a Terraform
```
Comment (bharathkkb) [inline README.md:16]: These reference links do no exist in the readme, we should add them.

File: README.md [hunk 2]
```diff
@@ -10,10 +10,12 @@ This module consists of the following submodules:
...
 0.12.x-compatible version of this module, the last released version
-intended for Terraform 0.12.x is [v4.5.0](https://registry.terraform.io/modules/terraform-google-modules/-sql-db/google/v4.5.0).
+intended for Terraform 0.12.x is [12.3.0].
```
Comment (bharathkkb) [inline README.md:18]: `12.3.0` is not a valid SQL module release. `v5.0.0` is the last 0.12.x compat release.

File: kitchen.yml [hunk 3]
```diff
@@ -40,11 +40,6 @@ suites:
       name: terraform
       root_module_directory: test/fixtures/postgresql-public
       command_timeout: 1800
-  - name: postgresql-public-iam
```
Comment (bharathkkb) [inline kitchen.yml:43]: Why are we removing `postgresql-public-iam`, I still see a test for this

**Review by bharathkkb — APPROVED** [review]
Minor nits, but overall LGTM

File: README.md (hunk 1 above)
Comment (g-awmalik) [reply README.md:16]: Done

File: README.md (hunk 2 above)
Comment (g-awmalik) [reply README.md:18]: Done.

File: kitchen.yml (hunk 3 above)
Comment (g-awmalik) [reply kitchen.yml:43]: There's an issue with the provider that was causing the test to fail. I've reported it here [#9862](https://github.com/hashicorp/terraform-provider-google/issues/9862). Once it goes live we can get this back in. Right now it's going to block the build.

Comment (bharathkkb) [reply kitchen.yml:43]: Thanks, let's open an issue in the repo to re add this back once upstream releases.

Comment (g-awmalik) [reply kitchen.yml:43]: https://github.com/terraform-google-modules/terraform-google-sql-db/issues/243

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**